File: leadMangement/src/main/java/com/management/lead/leadMangement/dto/SignUpRequest.java
package com.management.lead.leadMangement.dto;

import jakarta.validation.constraints.NotBlank;
import lombok.Data;

@Data
public class SignUpRequest {

    @NotBlank
    private String username;

    @NotBlank
    private String password;

    @NotBlank
    private String email;

    @NotBlank
    private String companyname;

    @NotBlank
    private String phone;

    private String role;

    @NotBlank
    private String teamname;
}

File: leadmanageclient/src/app/model/SignUpRequest.ts
export class SignUpRequest {
    username: string;
    password: string;
    email: string;
    companyname: string;
    phone: string;
    role: string;
    teamname: string;

    constructor(data: any = {}) {
        this.username = data.username || '';
        this.password = data.password || '';
        this.email = data.email || '';
        this.companyname = data.companyname || '';
        this.phone = data.phone || '';
        this.role = data.role || '';
        this.teamname = data.teamname || '';
    }
}
